docs/book/src/quinn/certificate.md

```diff
@@ -0,0 +1,192 @@
+You can change certificate validation logic when the `dangerous_configuration` feature flag of [rustls][3] is enabled.
+Then the only thing that needs to be done is to configure the client to trust any server.
+
+Start with adding a [rustls][3] dependency with the `dangerous_configuration` feature flag to your `Cargo.toml` file.
+
+```toml
+quinn = "*"
+rustls = { version = "*", features = ["dangerous_configuration", "quic"] }
+``` 
+
+Then, you can skip the certificate validation on the client by implementing [ServerCertVerifier][ServerCertVerifier] and let it assert true for any server. 
+
+```rust
+// Implementation of `ServerCertVerifier` that verifies everything as trustworthy.
+struct SkipCertificationVerification;
+
+impl rustls::ServerCertVerifier for SkipCertificationVerification {
+    fn verify_server_cert(
+        &self,
+        _roots: &rustls::RootCertStore,
+        _presented_certs: &[rustls::Certificate],
+        _dns_name: webpki::DNSNameRef,
+        _ocsp_response: &[u8],
+    ) -> Result<rustls::ServerCertVerified, rustls::TLSError> {
+        Ok(ServerCertVerified::assertion())
+    }
+}
+```
+
+After that, we can configure our [ClientConfig][ClientConfig] to use this new [ServerCertVerifier][ServerCertVerifier]. 
+
+```rust
+pub fn insecure() -> ClientConfig {
+    let mut cfg = quinn::ClientConfigBuilder::default().build();
+
+    // Get a mutable reference to the 'crypto' config in the 'client config'..
+    let tls_cfg: &mut rustls::ClientConfig =
+        std::sync::Arc::get_mut(&mut cfg.crypto).unwrap();
+
+    // Change the certification verifier.
+    // This is only available when compiled with 'dangerous_configuration' feature.
+    tls_cfg
+        .dangerous()
+        .set_certificate_verifier(Arc::new(SkipCertificationVerification));
+    cfg
+}
+```
+ 
+Finally, if you plug this [ClientConfig][ClientConfig] into the [EndpointBuilder::default_client_config()][default_client_config] your client endpoint should verify all connections as trustworthy.
+
+## Using Certificates
+
+In this section we look at certifying an endpoint with a real certificate. 
+This can be done with either a real certificate or a self-identified certificate.
+
+Let's define two useful functions that can dissect byte certificates and return quinn types.
+
+```rust
+pub fn parse_der(cert: Vec<u8>, private_key: Vec<u8>) -> anyhow::Result<(quinn::Certificate, quinn::PrivateKey)> {
+    let cert = quinn::Certificate::from_der(&cert)?;
+    let key = quinn::PrivateKey::from_der(&private_key)?;
+    Ok((cert, key))
+}
+
+pub fn parse_pem(cert: Vec<u8>, private_key: Vec<u8>) -> anyhow::Result<(quinn::Certificate, quinn::PrivateKey)> {
+    // Parse to certificate chain whereafter taking the first certifcater in this chain.
+    let cert = quinn::CertificateChain::from_pem(&cert)?.iter().next().unwrap().clone();
+    let key = quinn::PrivateKey::from_pem(&private_key)?;
+
+    Ok((quinn::Certificate::from(cert), key))
+}
+```
```

Comment:
I don't see much value in showing these kinds of utility functions in the documentations, since they're not at all relevant to Quinn itself.

Reply:
I think we can move this into the other methods.